modepy/spaces.py

```diff
@@ -88,12 +108,21 @@ def space_dim(self):
             return reduce(mul, range(order + 1, order + spdim + 1), 1) \
                     // reduce(mul, range(1, spdim + 1), 1)
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return (f"{type(self).__name__}("
             f"spatial_dim={self.spatial_dim}, order={self.order}"
             ")")
 
 
+@space_for_shape.register(Simplex)
+def _space_for_simplex(shape: Simplex, order: int):
+    return PN(shape.dim, order)
+
+# }}}
+
+
+# {{{ QN
+
 class QN(FunctionSpace):
```

Comment:
Should this inherit from `TensorProductSpace`?

Reply:
Yes! Made it that way somewhere in 05fe4d1 (that's why a lot of `register` for `QN` are gone).